ipf/src/main/java/org/motechproject/ipf/service/impl/IPFTemplateServiceImpl.java

```diff
@@ -0,0 +1,133 @@
+package org.motechproject.ipf.service.impl;
+
+import org.apache.commons.collections.map.HashedMap;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.ArrayUtils;
+import org.apache.commons.lang.StringUtils;
+import org.motechproject.ipf.domain.IPFTemplate;
+import org.motechproject.ipf.service.IPFTemplateDataService;
+import org.motechproject.ipf.service.IPFTemplateService;
+import org.motechproject.ipf.util.Constants;
+import org.motechproject.server.config.SettingsFacade;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.annotation.PostConstruct;
+import java.io.File;
+import java.io.FileFilter;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.regex.Pattern;
+
+@Service("ipfTemplateService")
+public class IPFTemplateServiceImpl implements IPFTemplateService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(IPFTemplateServiceImpl.class);
+    private static final String PROPERTIES_EXTENSION = "properties";
+
+    @Autowired
+    @Qualifier("ipfSettings")
+    private SettingsFacade settingsFacade;
+
+    @Autowired
+    private IPFTemplateDataService ipfTemplateDataService;
+
+    @PostConstruct
+    public void init() {
+        Properties properties = settingsFacade.getProperties(Constants.IPF_PROPERTIES_FILE);
```

Comment:
What are we exactly doing here? Isn't just storing templates in the database enough? Why are we reading files from the fs?